### PR TITLE
Rename launch-as-user to run-as-user

### PR DESCRIPTION
**Pull request by shamsimam**
Fixes issue #19

### CODE REVIEW

File: waiter/src/waiter/auth/authentication.clj [hunk 1]
```diff
@@ -29,10 +29,10 @@
 
 ;; An anonymous request does not contain any authentication information.
 ;; This is equivalent to granting everyone access to the resource.
-;; The anonymous authenticator attaches the principal of launch-as-user to the request.
-;; In particular, this enables requests to launch processes as launch-as-user.
+;; The anonymous authenticator attaches the principal of run-as-user to the request.
+;; In particular, this enables requests to launch processes as run-as-user.
 ;; Use of this authentication mechanism is strongly discouraged for production use.
-(defrecord AnonymousAuthenticator [launch-as-user]
+(defrecord AnonymousAuthenticator [run-as-user]
```
Comment (sradack) [inline waiter/src/waiter/auth/authentication.clj]: Can we go with SingleUserAuthenticator?

Comment (shamsimam) [inline waiter/src/waiter/auth/authentication.clj]: But it is not authenticating anyone. It is accepting requests from anyone and making them behave as a request from a specific user.

**Comment by shamsimam** [comment]
@sradack renamed to `SingleUserAuthenticator`

File: waiter/config-full.edn [hunk 2]
```diff
@@ -151,9 +151,9 @@
                         :kind :anonymous
```
Comment (dposada) [inline waiter/config-full.edn]: Maybe we should match cook and have the `:kind` be `:one-user` (https://github.com/twosigma/Cook/blob/master/scheduler/src/cook/components.clj#L316)

File: waiter/config-full.edn [hunk 3]
```diff
@@ -151,9 +151,9 @@
...
                         :anonymous {;; Custom implementations should specify a :factory-fn
                                     ;; that returns an instance of waiter.auth.authentication.Authenticator:
-                                    :factory-fn waiter.auth.authentication/anonymous-authenticator
+                                    :factory-fn waiter.auth.authentication/single-user-authenticator
```
Comment (dposada) [inline waiter/config-full.edn]: Same comment as above, maybe `one-user-authenticator` to match cook

**Comment by shamsimam** [comment]
@sradack any further feedback on this?

**Comment by sradack** [comment]
LGTM

**Pull request merged**